src/tye/Program.DeployCommand.cs

```diff
@@ -118,7 +118,9 @@ internal static void ApplyRegistry(OutputContext output, ApplicationBuilder appl
                 var registry = output.Prompt("Enter the Container Registry (ex: 'example.azurecr.io' for Azure or 'example' for dockerhub)", allowEmpty: !requireRegistry);
                 if (!string.IsNullOrWhiteSpace(registry))
                 {
-                    application.Registry = new ContainerRegistry(registry.Trim());
+                    var registryPullSecrets = output.Prompt("Enter the kubernetes credential secret to authorize pulling an image from the registry or leave empty", allowEmpty: true);
```

Comment:
I don't think we should prompt for pull secrets always via interactive. Interactive is only for required items, and pull secrets is more of an optional parameter.